src/jobs/sam-build.yml

```diff
@@ -6,7 +6,7 @@ parameters:
   pyenv-version:
     description: Version of python to use with pyenv
     type: string
-    default: "3.6.11"
+    default: "3.8.12"
```

Comment:
I think we already discussed that we wont touch the existing jobs but rather use their parameters to call them with a parameter value of our choice

We shouldn't be updating sam-build job's default python version direclty here as it can lead to backward compatibility issues for other consumers of this orb.

Rather, we should call the sam-build job with the pyenv-version parameter specified of our choice from the service using the orb.